src/Services/FHIR/FhirPatientService.php

```diff
@@ -54,32 +54,59 @@ public function createPatientResource($resourceId = '', $data = '', $encode = tr
 
     public function parsePatientResource($fhirJson)
     {
-        $data["date"] = $fhirJson['lastUpdated'];
-        $name = [];
-        foreach ($fhirJson["name"] as $sub_name) {
-            if ($sub_name["use"] == "official") {
-                $name = $sub_name;
-                break;
+        if (isset($fhirJson['name'])) {
+            $name = [];
+            foreach ($fhirJson["name"] as $sub_name) {
+                if ($sub_name["use"] == "official") {
+                    $name = $sub_name;
+                    break;
+                }
+            }
+            if (isset($name["family"])) {
+                $data["lname"] = $name["family"];
+            }
+            if ($name["given"][0]) {
+                $data["fname"] = $name["given"][0];
+            }
+            if (isset($name["given"][1])) {
+                $data["mname"] = $name["given"][1];
             }
         }
-        $data["lname"] = $name["family"];
-        $data["fname"] = $name["given"][0];
-        $data["mname"] = $name["given"][1];
-        $data["street"] = $fhirJson["address"][0]["line"][0];
-        $data["postal_code"] = $fhirJson["address"][0]["postalCode"];
-        $data["city"] = $fhirJson["address"][0]["city"];
-        $data["state"] = $fhirJson["address"][0]["state"];
-        $data["country_code"] = null;
-        $phone = [];
-        foreach ($fhirJson["telecom"] as $phone) {
-            if ($phone["use"] == "mobile") {
-                break;
+        if (isset($fhirJson["address"])) {
+            if (isset($fhirJson["address"][0]["line"][0])) {
+                $data["street"] = $fhirJson["address"][0]["line"][0];
+            }
+            if (isset($fhirJson["address"][0]["postalCode"][0])) {
+                $data["postal_code"] = $fhirJson["address"][0]["postalCode"];
+            }
+            if (isset($fhirJson["address"][0]["city"][0])) {
+                $data["city"] = $fhirJson["address"][0]["city"];
+            }
+            if (isset($fhirJson["address"][0]["state"][0])) {
+                $data["state"] = $fhirJson["address"][0]["state"];
+            }
+            if (isset($fhirJson["address"][0]["country"][0])) {
+                $data["country"] = $fhirJson["address"][0]["country"];
             }
         }
-        $data["phone_contact"] = $phone["value"];
-        $data["DOB"] = $fhirJson["birthDate"];
-        $data["sex"] = ucwords($fhirJson["gender"]);
-
+        if (isset($fhirJson["telecom"])) {
+            $phone = [];
+            foreach ($fhirJson["telecom"] as $phone) {
+                if ($phone["use"] == "mobile") {
+                    $name = $phone;
```

Comment:
This doesn't make sense! $name not used. Could just go ahead and set phone here.
Just fix when we add the rest of the possible phones

Reply:
BTW I didn't like the way we parse our FHIR resources. I think we could do better.
Ohno, One more task in the bucket list. My Bucket List is growing exponentially. And I am too lazy to pull requests. Will make a better schedule from now😤